### DISCUSSION TITLE
Access to betas (pre-releases) under HACS 2.0.0

### DISCUSSION THREAD

**Discussion by BJReplay** [BJReplay/ha-solcast-solar#148] [Announcements]
If you're you've upgraded HACS to 2.0.0 (and you really probably should - it looks a little different, but the update notifications for downloaded integrations, including manually installed integrations like Solcast PV Solar are a great addition) then enabling access to beta (pre-release) versions of integrations and dashboard (Lovelace) components has changed.

> Consider upgrading, unless you're affected the issues listed in the breaking changes for the [release](https://github.com/hacs/integration/releases/tag/2.0.0)):
> - You're running a version of Home Assistant earlier than 2024.4.1
> - You're using the sensor platform
> - You specifically need to configure HACS in YAML
> - You're using NetDaemon apps in HACS
> - You're using the /hacsfiles/ endpoint in themes
> - You're using hacs prefixed events
> - You're using a HACS hosted integration or dashboard component and absolutely depend on info.md file being rendered in HACS

Previously you could enable access to beta releases (releases tagged as _pre-release_ in GitHub) from the download dialog for an integration (there was a toggle switch in the download dialog), and this would allow you to see pre-release versions as well as release versions in the list of versions that you could download - but it wasn't always possible to choose a specific version to download.  You could choose a version, but HACS couldn't always swap to an earlier version.

Now HACS warns against using the download version mechanism to swap to an earlier version but does seem to do a pretty good job of reverting to an earlier version if you choose to use the new dialog to choose an earlier version.

![image](https://github.com/user-attachments/assets/d983b23d-5e25-40bd-b86c-889d7d7312d2)

It also shows you which versions are tagged as pre-releases:

![image](https://github.com/user-attachments/assets/cd5b2eb7-3749-44bb-beeb-4a34291e40ec)

However, to enable access to pre-releases, you have to enable a switch entity that is disabled by default, wait 30 seconds for Home Assistant to make that entity available in the user interface, and then turn it on.

Once you've turned it on, HACS will then download pre-release versions for a particular HACS integration if they are newer than the latest release version.

We're shortly going to be releasing a beta (pre-release) version, and calling for beta testers - when we do, we'll be referring to this discussion post to let you know how to access the pre-release.

The easiest way to turn on that switch for Solcast PV Solar is to search your entities for `switch.solcast`: Click the link below (or go to Settings, Devices & Services, Entities) to go your entities, and filter the list with `switch.solcast` to find the disabled entity you need to enable.

[![image](https://github.com/user-attachments/assets/e673531a-eb8a-48a8-9b3f-5930b2bfdb47)](https://my.home-assistant.io/redirect/entities/)

By default, the switch will be disabled:
![image](https://github.com/user-attachments/assets/04274f11-1e14-4026-9ecc-be19923378b1)

Click on the entity where you will see that it is disabled

![image](https://github.com/user-attachments/assets/63c5678c-b85c-498b-b0c6-5bf68db93867)

Then click on the gear icon which will expand the dialog, and you will be able to click on ENABLE in the warning bar or toggle the Enable switch.

![image](https://github.com/user-attachments/assets/26fedc6c-49bc-4c48-970b-5135148749e7)

You'll see a message indicating that the entities will be added to Home Assistant in 30 seconds.

![image](https://github.com/user-attachments/assets/18a5bf16-a3b1-46a4-adb9-076be3651c38)

Click on OK and twiddle your thumbs.  It **will** take 30 seconds.

If you left the previous dialog open, instead of having a disabled entity visible, you've now got a switch. If not, you can open it - it looks like this:

![image](https://github.com/user-attachments/assets/f0f3273f-cb90-448c-b407-72c737c99434)

Flick it on, and you'll be set to get pre-release versions;

![image](https://github.com/user-attachments/assets/0c9350d7-4d3d-4d4b-a668-deca27b69d08)

You can then wait (if you're doing this ahead of time, because you're a keen beta tester), or you can go back to HACS, select Solcast PV Solar, select the vertical ellipsis, and choose Update information which will probably cause HACS to prompt for an update if there's a new version, or Redownload which should grab the latest version, including any pre-release.

**Comment by gcoan** [comment]
Great instructions thanks @BJReplay, bit of weird behaviour for my setup.

The pre-release switch was already enabled and turned on for me (presumably due to previously using beta versions in HACS), but when 4.1.5 was released, I didn't get an update notification despite waiting more than a day for it to arrive.

In my entities I had switch.solcast_pv_solar_pre_release turned on, a sensor update.solcast_pv_solar_update that was status 'unavailable and no longer being supplied' and update.solcast_pv_solar_update_2 that was available but not showing any update.

I guess the duality of the _update sensor is a hangover from when I changed over to the BJreplay version?  But it clearly wasn't working.

Deleted update.solcast_pv_solar_update entity, renamed _update_2 to _update and straight away I got the notification that there was version 4.1.5 ready to download.
